File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/60968-Visual Studio 2010 SDK Samples/VSSDK IDE Sample Combo Box/C#/TDD/TestVsPkg.cs
/***************************************************************************

Copyright (c) Microsoft Corporation. All rights reserved.
This code is licensed under the Visual Studio SDK license terms.
THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

***************************************************************************/

// System references.
using System;
using System.Reflection;

// Platform references
using Microsoft.VisualStudio.Shell.Interop;

// Unit Test framework
using Microsoft.VsSDK.UnitTestLibrary;
using Microsoft.VisualStudio.TestTools.UnitTesting;

// Assembly to test
using Microsoft.Samples.VisualStudio.ComboBox;

namespace Microsoft.Samples.VisualStudio.ComboBox.UnitTest {

    [TestClass()]
    public class ComboBoxPackageTest {

        [TestMethod()]
        public void PackageCreation() {
            ComboBoxPackage package = new ComboBoxPackage();
            Assert.IsNotNull(package, "Failed to create package");
        }

        [TestMethod()]
        public void PackageInitialize() {
            ComboBoxPackage packageObject = new ComboBoxPackage();
            Assert.IsNotNull(packageObject, "Failed to create package");
            IVsPackage package = packageObject as IVsPackage;
            Assert.IsNotNull(package, "Can not get the package interface from the package object.");
            using (OleServiceProvider serviceProvider = OleServiceProvider.CreateOleServiceProviderWithBasicServices())
            {
                Assert.AreEqual(0, package.SetSite(serviceProvider), "Can not get the package interface from the package object.");
            }
        }

        [TestMethod()]
        public void PackageInitializeWithNull()
        {
            ComboBoxPackage packageObject = new ComboBoxPackage();
            Assert.IsNotNull(packageObject, "Failed to create package");
            IVsPackage package = packageObject as IVsPackage;
            Assert.IsNotNull(package, "Can not get the package interface from the package object.");
            Assert.AreEqual(0, package.SetSite(null), "SetSite returned an unexpected value");
        }

        [TestMethod()]
        public void PackageResourcesCreation()
        {
            System.Type t = System.Type.GetType("Microsoft.Samples.VisualStudio.ComboBox.Resources, Reference.ComboBox");
            object resources = System.Activator.CreateInstance(t, true);

            Assert.IsNotNull(resources, "Failed to create resources");

            MethodInfo method_get_Culture = t.GetMethod("get_Culture", BindingFlags.NonPublic | BindingFlags.Static);
            object result1 = method_get_Culture.Invoke(null, null);

            MethodInfo method_set_Culture = t.GetMethod("set_Culture", BindingFlags.NonPublic | BindingFlags.Static);
            object result2 = method_set_Culture.Invoke(null, new object[] { System.Globalization.CultureInfo.CurrentCulture });
        }
    }
}

File: Visual Studio Product Team/Visual Studio 2010 SDK Samples/60968-Visual Studio 2010 SDK Samples/VSSDK IDE Sample Combo Box/VB/TDD/TestVsPkg.vb
'**************************************************************************

'Copyright (c) Microsoft Corporation. All rights reserved.
'This code is licensed under the Visual Studio SDK license terms.
'THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
'ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
'IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
'PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.

'**************************************************************************

' System references.

Imports Microsoft.VisualBasic
Imports System
Imports System.Reflection

' Platform references.
Imports Microsoft.VisualStudio.Shell.Interop

' Unit Test framework.
Imports Microsoft.VsSDK.UnitTestLibrary
Imports Microsoft.VisualStudio.TestTools.UnitTesting

' Assembly to test.
Imports Microsoft.Samples.VisualStudio.ComboBox

Namespace Microsoft.Samples.VisualStudio.ComboBox.UnitTest

	<TestClass()> _
	Public Class ComboBoxPackageTest

		<TestMethod()> _
		Public Sub PackageCreation()
            Dim package As New ComboBoxPackage()
			Assert.IsNotNull(package, "Failed to create package")
		End Sub

		<TestMethod()> _
		Public Sub PackageInitialize()
            Dim packageObject As New ComboBoxPackage()
			Assert.IsNotNull(packageObject, "Failed to create package")
			Dim package As IVsPackage = TryCast(packageObject, IVsPackage)
			Assert.IsNotNull(package, "Can not get the package interface from the package object.")
			Using serviceProvider As OleServiceProvider = OleServiceProvider.CreateOleServiceProviderWithBasicServices()
				Assert.AreEqual(0, package.SetSite(serviceProvider), "Can not get the package interface from the package object.")
			End Using
		End Sub

		<TestMethod()> _
		Public Sub PackageInitializeWithNull()
            Dim packageObject As New ComboBoxPackage()
			Assert.IsNotNull(packageObject, "Failed to create package")
			Dim package As IVsPackage = TryCast(packageObject, IVsPackage)
			Assert.IsNotNull(package, "Can not get the package interface from the package object.")
			Assert.AreEqual(0, package.SetSite(Nothing), "SetSite returned an unexpected value")
		End Sub

		<TestMethod()> _
		Public Sub PackageResourcesCreation()
            Dim t As System.Type = System.Type.GetType("Resources, Reference.ComboBox")
			Dim resources As Object = System.Activator.CreateInstance(t, True)

			Assert.IsNotNull(resources, "Failed to create resources")

			Dim method_get_Culture As MethodInfo = t.GetMethod("get_Culture", BindingFlags.NonPublic Or BindingFlags.Static)
			Dim result1 As Object = method_get_Culture.Invoke(Nothing, Nothing)

			Dim method_set_Culture As MethodInfo = t.GetMethod("set_Culture", BindingFlags.NonPublic Or BindingFlags.Static)
			Dim result2 As Object = method_set_Culture.Invoke(Nothing, New Object() { System.Globalization.CultureInfo.CurrentCulture })
		End Sub
	End Class
End Namespace
